### PR TITLE
Generate binaries

### PR DESCRIPTION
**Pull request by juliankoh**
### Motivation
Generate binaries to avoid having to compile manually. 

### Solution
- Uses a [fork](https://github.com/techknowlogick/xgo) of the [original xgo](https://github.com/karalabe/xgo) library to generate binaries for various platforms. This fork is used because it is better maintained and [avoids issues with compiling from local paths](https://github.com/karalabe/xgo/issues/115) 
- `make compile` command to compile for windows, linux and darwin.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -379,6 +379,7 @@ Global Flags:
 * `make test` to run tests
 * `make lint` to lint the source code (included generated code)
 * `make release` to run one last check before opening a PR
+* `make compile version=RELEASE_TAG` to generate binaries
```
Comment (patrick-ogrady) [inline README.md:382]: Would you mind adding a small section to the README about using binaries and that a script is in progress?

Comment (patrick-ogrady) [reply README.md:382]: You may want to add that it is a known issue to click download on the release:
https://github.com/hashicorp/terraform/issues/23033

Comment (patrick-ogrady) [reply README.md:382]: Turns out curl bypasses this:
```
curl -L https://github.com/coinbase/rosetta-cli/releases/download/v0.4.1/rosetta-cli-0.4.1-darwin-10.6-amd64 -o rosetta-cli; chmod +x rosetta-cli;
```

Comment (patrick-ogrady) [reply README.md:382]: You may want to include this in the README ^^ with a warning to not download on the UI

File: README.md [hunk 2]
```diff
@@ -29,10 +29,18 @@ Before diving into the CLI, we recommend taking a look at the Rosetta API Docs:
...
 
 ## Install
+To install `rosetta-cli` by getting the source code, run:
 ```
 go get github.com/coinbase/rosetta-cli
 ```
 
+To download the binary directly, run:
+```
+curl -L https://github.com/coinbase/rosetta-cli/releases/download/v0.4.1/rosetta-cli-0.4.1-darwin-10.6-amd64 -o rosetta-cli; chmod +x rosetta-cli;
```
Comment (patrick-ogrady) [inline README.md:39]: You may want to do a `rosetta-cli-{YOUR REQUIRED VERSION HERE}`

File: README.md [hunk 3]
```diff
@@ -29,10 +29,18 @@ Before diving into the CLI, we recommend taking a look at the Rosetta API Docs:
 * [Construction API](https://www.rosetta-api.org/docs/construction_api_introduction.html)
 
 ## Install
+To install `rosetta-cli` by getting the source code, run:
 ```
 go get github.com/coinbase/rosetta-cli
 ```
 
+To download the binary directly, run:
```
Comment (patrick-ogrady) [inline README.md:37]: nit: `a binary`

File: README.md [hunk 4]
```diff
@@ -29,10 +29,18 @@ Before diving into the CLI, we recommend taking a look at the Rosetta API Docs:
 * [Construction API](https://www.rosetta-api.org/docs/construction_api_introduction.html)
 
 ## Install
+To install `rosetta-cli` by getting the source code, run:
```
Comment (patrick-ogrady) [inline README.md:32]: not: To install `rosetta-cli` from source, run:

File: README.md [hunk 5]
```diff
@@ -29,10 +29,18 @@ Before diving into the CLI, we recommend taking a look at the Rosetta API Docs:
 * [Construction API](https://www.rosetta-api.org/docs/construction_api_introduction.html)
 
 ## Install
+To install `rosetta-cli` from source, run:
 ```
 go get github.com/coinbase/rosetta-cli
 ```
 
+To download a binary directly for MacOS, run:
```
Comment (patrick-ogrady) [inline README.md:37]: Do we want to list installs for each OS? I was thinking do:
```
rosetta-cli-{YOUR REQUIRED VERSION HERE}-{YOUR OS HERE}
```

And then include a link to releases to identify

And then say, for example on mac with version 0.4.1:
```
curl -L https://github.com/coinbase/rosetta-cli/releases/download/v0.4.1/rosetta-cli-0.4.1-darwin-10.6-amd64 -o rosetta-cli; chmod +x rosetta-cli;
```

File: Makefile [hunk 6]
```diff
@@ -49,6 +49,9 @@ salus:
 
 release: add-license shorten-lines format test lint salus
 
+compile:
+	xgo --targets=darwin/*,windows/*,linux/* -out bin/rosetta-cli-$(version) .
```
Comment (patrick-ogrady) [inline Makefile:53]: Is it possible to specify a tag here or do we need to checkout the code we want to compile @juliankoh ?

Comment (juliankoh) [reply Makefile:53]: This builds the code locally (as you can see from the . )

We can also build a specific tag but that would be a different command.

**Pull request merged**